test/python/quantum_info/test_random.py

```diff
@@ -25,9 +26,26 @@
 class TestRandomUtils(QiskitTestCase):
     """Testing qiskit.quantum_info.random.utils"""
 
+    def test_seeded_random_unitary(self):
+        """Given the same seed, the output should be expected.
+        Following the numpy approach we are going to keep the policy of platform-independence
+        with randomness.
+        https://stackoverflow.com/questions/40676205/cross-platform-numpy-random-seed"""
+        seed = 314159
+        unitary = random_unitary(4, seed=seed)
+        expected = np.array([[0.07749745 + 0.65152329j, 0.40784587 - 0.36593286j,
+                              0.03129288 + 0.2530866j, 0.36101855 - 0.27184549j],
+                             [0.53016612 - 0.01919937j, 0.48298729 + 0.19460984j,
+                              -0.40381565 - 0.53289147j, -0.00508656 + 0.01841975j],
+                             [-0.32716712 - 0.11010826j, -0.29569725 - 0.11013661j,
+                              -0.23722636 - 0.47383058j, 0.40993924 - 0.57656653j],
+                             [0.04033726 - 0.4089958j, 0.39452908 + 0.4163953j,
+                              0.37036483 + 0.26451022j, 0.08430501 - 0.53648299j]])
+        assert_allclose(unitary.data, expected, atol=1e-8)
```

Comment:
These sort of tests for a `random_unitary` function don't make a lot of sense and are sensitive to changes in their implementation.

For the `random_unitary` function for example the main coverage you would want in tests is:

- **Test seed:** run function with fixed seed multiple time and check output doesn't change
- **Test is_unitary:** run function and test output is a unitary matrix of the correct dimension
- **Test distribution:** you test that averaging over many random unitaries converges towards the expected average  (this one is a bit more complicated and we don't have coverage of yet iirc)

Similarly for random states vectors and random density matrices (but checking valid instead of unitary)

Reply:
This is solved by https://github.com/Qiskit/qiskit-terra/pull/4058 . Thanks!